### PR TITLE
[video][android] Add Picture in Picture functionality

### PR DESCRIPTION
**Pull request by behenate**
# Why

This PR adds PiP mode to expo video.  Unlike iOS, where only the video view enters PiP, on Android entering PiP makes the entire app enter PiP mode. We can experiment with iOS-like implementation in the future, but for now this is more in line with most of the native PiP implementations on Android (eg. the YouTube app).

Also added `isPictureInPictureSupportedAsync` function for both Android and iOS, which should be helpful for the users.

# How

Used `Activity.enterPictureInPicture` to make the `MainActivity` enter PiP. Some tweaking and changing of the layout after entering PiP was necessary in order to make this functionality work well.

Modified the `expo-video` config plugin to add `supportsPictureInPicture` to `MainActivity` in `AndroidManifest`

For now `enterPiPAutomatically` prop works only on Android 12+. We will need some changes to React Native's codebase in order to make it work on older versions of Android.

# Test Plan

Tested in BareExpo on a real Android 13 device

https://github.com/expo/expo/assets/31368152/f0e58417-2188-43c8-bf28-7f562e4670f9

### CODE REVIEW

**Review by lukmccall — APPROVED** [review]

File: packages/expo-video/plugin/src/withExpoVideo.ts [hunk 1]
```diff
@@ -1,4 +1,28 @@
-import { ConfigPlugin, withInfoPlist } from '@expo/config-plugins';
+import { ConfigPlugin, withInfoPlist, withAndroidManifest } from '@expo/config-plugins';
+import { Manifest } from '@expo/config-plugins/build/android';
```
Comment (EvanBacon) [inline packages/expo-video/plugin/src/withExpoVideo.ts:2]:
```suggestion
import { AndroidConfig, ConfigPlugin, withInfoPlist, withAndroidManifest } from '@expo/config-plugins';
const { Manifest } = AndroidConfig;
```

Don't use the nested imports.

File: packages/expo-video/plugin/src/withExpoVideo.ts [hunk 2]
```diff
@@ -9,6 +33,20 @@ const withExpoVideo: ConfigPlugin = (config) => {
...
+      console.error("Couldn't apply the expo-video plugin for Android");
+      return config;
+    }
+
+    manifest.application.forEach(enablePiPSupport);
+
+    config.modResults.manifest = manifest;
+
+    return config;
+  });
```
Comment (EvanBacon) [inline packages/expo-video/plugin/src/withExpoVideo.ts:49]: Probably just something simple like this:

```suggestion
  withAndroidManifest(config, (config) => {
    const activity = AndroidConfig.Manifest.getMainActivityOrThrow(config.modResults);
    activity.$['android:supportsPictureInPicture'] = 'true';
    return config;
  });
```

File: packages/expo-video/src/VideoView.tsx [hunk 3]
```diff
@@ -24,15 +33,15 @@ export class VideoView extends PureComponent<VideoViewProps> {
   /**
    * Enters Picture in Picture (PiP) mode. Throws an exception if the device does not support PiP.
    * > **Note:** Only one player can be in Picture in Picture (PiP) mode at a time.
-   * @platform ios 14+
+   * @platform android, ios 14+
```
Comment (Simek) [inline packages/expo-video/src/VideoView.tsx:36]:
```suggestion
   * @platform android
   * @platform ios 14+
```

Currently, to mark method correctly as supported by multiple platforms, we need to use separate `@platform` tag:
* https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md#general-approach

File: packages/expo-video/plugin/src/withExpoVideo.ts (hunk 2 above)
Comment (behenate) [reply packages/expo-video/plugin/src/withExpoVideo.ts:49]: Wow that's so much simpler! Thanks 😄

**Review by tsapeta — APPROVED** [review]

**Pull request merged**